sdk/communication/Azure.Communication.Rooms/src/RoomsClient.cs

```diff
@@ -0,0 +1,511 @@
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Azure.Communication.Pipeline;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Communication.Rooms
+{
+    /// <summary>
+    /// The Azure Communication Services Rooms client.
+    /// </summary>
+    public class RoomsClient
+    {
+        private readonly ClientDiagnostics _clientDiagnostics;
+
+        internal RoomsRestClient RoomsServiceClient { get; }
+
+        #region public constructors - all arguments need null check
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RoomsClient"/> class.
+        /// </summary>
+        /// <param name="connectionString"></param>
+        public RoomsClient(string connectionString)
+            : this(
+                ConnectionString.Parse(Argument.CheckNotNullOrEmpty(connectionString, nameof(connectionString))),
+                new RoomsClientOptions())
+        { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RoomsClient"/> class.
+        /// </summary>
+        /// <param name="connectionString"></param>
+        /// <param name="options"></param>
+        public RoomsClient(string connectionString, RoomsClientOptions options)
+            : this(
+                ConnectionString.Parse(Argument.CheckNotNullOrEmpty(connectionString, nameof(connectionString))),
+                options ?? new RoomsClientOptions())
+        { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RoomsClient"/> class.
+        /// </summary>
+        /// <param name="endpoint"></param>
+        /// <param name="keyCredential"></param>
+        /// <param name="options"></param>
+        public RoomsClient(Uri endpoint, AzureKeyCredential keyCredential, RoomsClientOptions options = default)
+            : this(
+                Argument.CheckNotNull(endpoint, nameof(endpoint)).AbsoluteUri,
+                Argument.CheckNotNull(keyCredential, nameof(keyCredential)),
+                options ?? new RoomsClientOptions())
+        { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RoomsClient"/> class.
+        /// </summary>
+        /// <param name="endpoint"></param>
+        /// <param name="tokenCredential"></param>
+        /// <param name="options"></param>
+        public RoomsClient(Uri endpoint, TokenCredential tokenCredential, RoomsClientOptions options = default)
+            : this(
+                Argument.CheckNotNull(endpoint, nameof(endpoint)).AbsoluteUri,
+                Argument.CheckNotNull(tokenCredential, nameof(tokenCredential)),
+                options ?? new RoomsClientOptions())
+        { }
+
+        #endregion
+
+        #region private constructors
+
+        private RoomsClient(ConnectionString connectionString, RoomsClientOptions options)
+            : this(connectionString.GetRequired("endpoint"), options.BuildHttpPipeline(connectionString), options)
```

Comment:
What exception message does this throw if the value isn't found?

Reply:
If endpoint is not found in connectionstring it throws InvalidOperationException

Snipped from Shared/ConnectionString.cs
public string GetRequired(string keyword) =>
            _pairs.TryGetValue(keyword, out var value) ? value : throw new InvalidOperationException($"Required keyword '{keyword}' is missing in connection string.");